dali/operators/random/rng_base_gpu.h

```diff
@@ -27,23 +27,19 @@
 
 namespace dali {
 
-template <bool IsNoiseGen>
-struct BlockDesc;
-
-template <>
-struct BlockDesc<false> {
-  int sample_idx;
-  void* output;
-  size_t size;
+struct SampleDesc {
```

Comment:
Divided into BlockDesc and SampleDesc, because we added some info that doesn't need to be repeated for every block (pixel stride, channel stride, number of channels...)